modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/TransportMultiSearchTemplateAction.java

```diff
@@ -76,6 +85,16 @@ protected void doExecute(Task task, MultiSearchTemplateRequest request, ActionLi
                 searchRequest = convert(searchTemplateRequest, searchTemplateResponse, scriptService, xContentRegistry, searchUsageHolder);
             } catch (Exception e) {
                 items[i] = new MultiSearchTemplateResponse.Item(null, e);
+                if (ExceptionsHelper.status(e).getStatus() >= 500 && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                    try {
+                        String requestInfo = Strings.toString(
+                            searchTemplateRequest.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS)
+                        );
+                        logger.warn("MultiSearchTemplate convert failure for request " + requestInfo, e);
+                    } catch (IOException ex) {
+                        logger.warn("MultiSearchTemplate convert failure for request (unable to convert request to JSON)" + ex, e);
+                    }
+                }
                 continue;
             }
             items[i] = new MultiSearchTemplateResponse.Item(searchTemplateResponse, null);
```

Comment:
I'm curious to understand: the above should catch cases where the template conversion already causes an exception that we would later deliver in an item. Those would come e.g. from the script engine, but there doesn't seem to be any network traffic involved?
What about executing the actual multiSearchRequest here later? Are those cases covered by the other logging site added here in TransportMultiSearchAction? I think so but wanted to make sure I'm not missing something.

Reply:
The template action goes through the TransportMultiSearchAction for the actual searching, so it will get the logging added there (that's why I combined these two into one PR).

I noticed that the template action has the additional "convert" step that just catches exceptions, ignores them and continues. Since that convert could throw NPEs or other things that indicate a code bug, I decided to add logging for them as well.